contracts/CredentialAtomicQueryMTP.sol

```diff
@@ -1,7 +1,159 @@
 // SPDX-License-Identifier: GPL-3.0
 pragma solidity ^0.8.0;
 
-import {GenesisUtils} from "./lib/GenesisUtils.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "solidity-bytes-utils/contracts/BytesLib.sol";
+
+library GenesisUtils {
```

Comment:
i would define a different file for this contract, i think it's clearer and more readable ^^

Reply:
fixed